lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3167,12 +3168,40 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
     attr->setVJPFunction(vjp);
   }
 
-  auto insertion =
-      ctx.DifferentiableAttrs.try_emplace({D, checkedWrtParamIndices}, attr);
+  if (auto *asd = dyn_cast<AbstractStorageDecl>(D)) {
+    // Remove `@differentiable` attribute from storage declaration to prevent
+    // duplicate attribute registration during SILGen.
+    D->getAttrs().removeAttribute(attr);
+    // Transfer `@differentiable` attribute from storage declaration to
+    // getter accessor.
+    auto *newAttr = DifferentiableAttr::create(
```

Comment:
Note: creating a new attribute here for the getter seems necessary.
I tried reusing `attr` (removing it from the storage declaration `asd` and adding it to `asd->getGetter()`), but that led to problems.